pkg/controller/replication_controller.go

```diff
@@ -204,6 +204,10 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 // When a pod is created, enqueue the controller that manages it and update it's expectations.
 func (rm *ReplicationManager) addPod(obj interface{}) {
 	pod := obj.(*api.Pod)
+	if pod.DeletionTimestamp != nil {
```

Comment:
It's cleared during creation - that said, I'm making an additional guard to prevent deletion grace period seconds from being reset and will move your comment.

----- Original Message -----
> > @@ -204,6 +204,10 @@ func (rm *ReplicationManager) getPodControllers(pod
> > *api.Pod) *api.ReplicationCo
> >  // When a pod is created, enqueue the controller that manages it and
> >  update it's expectations.
> >  func (rm *ReplicationManager) addPod(obj interface{}) {
> >  	pod := obj.(*api.Pod)
> > +	if pod.DeletionTimestamp != nil {
> 
> The comment is better placed above updatePod, pologies for not mentioning
> this.
> 
> If one can create an rc with a pod template that has the deletion timestamp
> set, it will just keep creating pods. Are we disallowing this in validation
> or is it intended?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/8641/files#r31478081